script/test/utils.sh

```diff
@@ -312,25 +312,58 @@ test_setup() {
   run_crictl
 }
 
-# test_teardown kills containerd.
+# test_teardown kills containerd and associated shim processes, unmounts shm, and cleans up state.
 test_teardown() {
   if [ -n "${pid}" ]; then
     if [ $IS_WINDOWS -eq 1 ]; then
-      # Mark service for deletion. It will be deleted as soon as the service stops.
+      # Windows cleanup
       sc.exe delete containerd-test
       # Stop the service
       sc.exe stop containerd-test || true
     else
+      echo "[teardown] Stopping containerd for PID ${pid}..."
+
+      # Kill containerd process group
       pgid=$(ps -o pgid= -p "${pid}" || true)
-      if [ ! -z "${pgid}" ]; then
-        ${sudo} pkill -g ${pgid}
+      if [ -n "${pgid}" ]; then
+        ${sudo} pkill -g "${pgid}" || true
       else
-        echo "pid(${pid}) not found, skipping pkill"
+        echo "[teardown] pid(${pid}) not found, skipping pkill"
+      fi
```

Comment:
If the error from pkill is being ignored, why not just run pkill outright (avoids a potential TOCTTOU issue that existed before and now after the change)?